gradio/reload.py

```diff
@@ -3,34 +3,59 @@
 Contains the functions that run when `gradio` is called from the command line. Specifically, allows
 
 $ gradio app.py, to run app.py in reload mode where any changes in the app.py file or Gradio library reloads the demo.
-$ gradio app.py my_demo.app, to use variable names other than "demo"
+$ gradio app.py my_demo, to use variable names other than "demo"
 """
 import inspect
 import os
+import re
+import subprocess
 import sys
 from pathlib import Path
 
-from uvicorn import Config
-from uvicorn.supervisors import ChangeReload
-
 import gradio
-from gradio import networking, utils
+from gradio import utils
 
 
 def _setup_config():
     args = sys.argv[1:]
     if len(args) == 0:
         raise ValueError("No file specified.")
     if len(args) == 1 or args[1].startswith("--"):
-        demo_name = "demo.app"
+        demo_name = "demo"
     else:
         demo_name = args[1]
-        if "." not in demo_name:
+        if "." in demo_name:
+            demo_name = demo_name.split(".")[0]
             print(
-                "\nWARNING: As of Gradio 3.31, the parameter after the file path must be the name of the FastAPI app, not the Gradio demo. In most cases, this just means you should add '.app' after the name of your demo, e.g. 'demo' -> 'demo.app'."
+                "\nWARNING: As of Gradio 3.41.0, the parameter after the file path must be the name of the Gradio demo, not the FastAPI app. In most cases, this just means you should remove '.app' after the name of your demo, e.g. 'demo.app' -> 'demo'."
```

Comment:
```suggestion
                "\nWARNING: As of Gradio 3.42.0, the parameter after the file path must be the name of the Gradio demo, not the FastAPI app. In most cases, this just means you should remove '.app' after the name of your demo, e.g. 'demo.app' -> 'demo'."
```

Reply:
We should also update this guide here: https://www.gradio.app/guides/developing-faster-with-reload-mode